modules/codepipeline/main.tf

```diff
@@ -0,0 +1,69 @@
+## TODO: DeletionPolicy = Retain
+resource "aws_s3_bucket" "codepipeline" {
+  bucket = "artifact_store"
```

Comment:
probably want this as configurable in case of preexisting buckets with that name

Reply:
Good thought, will do.  There are some things like this that I think cloudformation just silently set that are required in terraform.